api/openapi.yaml

```
@@ -0,0 +1,115 @@
+openapi: 3.0.3
+
+info:
+  title: CSES API
+  description: API describing the communication between CSES CLI and cses.fi
+  version: 0.0.1
+  license:
+    name: MIT
+    url: https://github.com/H4m5t3r/cses-cli/blob/main/LICENSE
+
+
+
+
+components:
+  schemas:
+    ErrorMessage:
+      additionalProperties: false
+      required:
+        - message
+      type: object
+      properties:
+        message:
+          description: Error message from the server.
+          type: string
+
+  # define possible authentication methods
+  securitySchemes:
+    apiKeyAuth: # could be any name
+      type: apiKey
+      in: header
+      name: X-Auth-Token
+
+security:
+  - apiKeyAuth: []
+
+paths:
+  /login:
+    post:
+      # unique id for mocking
+      operationId: app.login_get
+
+      summary: Logs in the user with CSES credentials
+      description: |
```

(Nitpick: for consistency, use `>` on all description properties. There is likely no difference in the generated documentation though.)

```suggestion
      description: >
```